week05/Routers/Routers/Network.cs

```diff
@@ -0,0 +1,170 @@
+// <copyright file="Network.cs" company="SPBU">
+// Copyright (c) Alexander Bugaev 2024. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+// </copyright>
+
+namespace Routers;
+
+/// <summary>
+/// Network of connected routers.
+/// </summary>
+public class Network
+{
+    private int numberOfRouters;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Network"/> class.
+    /// Creates a network based on configuration from the file at the given path.
+    /// </summary>
+    /// <param name="filePath">File to read configuration from.</param>
+    public Network(string filePath)
+    {
+        this.Connections = new List<Connection>();
+        var lines = this.GetLines(filePath);
+        foreach (var line in lines)
+        {
+            this.AddConnections(line);
+        }
+    }
+
+    /// <summary>
+    /// Gets the list of channels in this network.
+    /// </summary>
+    public List<Connection> Connections { get; private set; }
+
+    /// <summary>
+    /// Write the configuration to the file in format: {router1}: {router2} {capacity}, ...
+    /// </summary>
+    /// <param name="filePath">Path of the file to write to.</param>
+    public void WriteConfiguration(string filePath)
+    {
+        var adjacentRouters = this.GetAdjacencyDict();
+        using StreamWriter writer = new StreamWriter(filePath, false);
+        foreach (var router in adjacentRouters.Keys)
+        {
+            writer.Write($"{router}:");
+            for (int i = 0; i < adjacentRouters[router].Count; ++i)
```

Comment:
```suggestion
            for (var i = 0; i < adjacentRouters[router].Count; ++i)
```
используй var везде, где можно :)